src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.Accessibility.cs

```diff
@@ -0,0 +1,392 @@
+            string source = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    private protected class Nested
+    {
+        [ModuleInitializer]
+        public static void M() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics(
+                // (8,10): error CS8794: Module initializer method 'M' must be accessible outside top-level type 'C'
+                //         [ModuleInitializer]
+                Diagnostic(ErrorCode.ERR_ModuleInitializerMethodMustBeAccessibleOutsideTopLevelType, "ModuleInitializer").WithArguments("M", "C").WithLocation(8, 10)
+                );
+        }
+
+        [Fact]
+        public void NestedTypeVisibilityMayBePublic()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    public class Nested
+    {
+        [ModuleInitializer]
+        public static void M() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void NestedTypeVisibilityMayBeInternal()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    internal class Nested
+    {
+        [ModuleInitializer]
+        public static void M() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void NestedTypeVisibilityMayBeProtectedInternal()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    protected internal class Nested
+    {
+        [ModuleInitializer]
+        public static void M() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void MayBeDeclaredByStruct()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+struct S
+{
+    [ModuleInitializer]
+    internal static void M() { }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void MayBeDeclaredByInterface()
```

Comment:
>MayBeDeclaredByInterface [](start = 20, length = 24)

This test should be success case, we should execute and observe expected behavior.

Reply:
This test demonstrates the success case.

Reply:
> This test demonstrates the success case.

Same response as for the next unit-test

---
In reply to: [418183951](https://github.com/dotnet/roslyn/pull/43301#discussion_r418183951) [](ancestors = 418183951)